File: module-1/11_Inheritance/exercise/src/main/java/com/techelevator/challenge/BankAccount.java
package com.techelevator.challenge;


import java.math.BigDecimal;

public class BankAccount {

    private BigDecimal balance;
    private String accountHolderName;
    private String accountNumber;

    public BankAccount(String accountHolderName, String accountNumber) {
        this(accountHolderName, accountNumber, new BigDecimal(0));
    }

    public BankAccount(String accountHolderName, String accountNumber, BigDecimal balance) {
        this.accountHolderName = accountHolderName;
        this.accountNumber = accountNumber;
        this.balance = balance;
    }

    //getters & setters:

    public String getAccountHolderName() {
        return accountHolderName;
    }
    public String getAccountNumber() {
        return accountNumber;
    }
    public BigDecimal getBalance() {
        return balance;
    }

    //other methods

    public BigDecimal deposit(BigDecimal amountToDeposit) {
        if (amountToDeposit.compareTo(new BigDecimal(0)) > 0) {
            balance.add(amountToDeposit);
        }
        return balance;
    }

    public BigDecimal withdraw(BigDecimal amountToWithdraw) {
        if(amountToWithdraw.compareTo(balance) <= 0) {
            balance.subtract(amountToWithdraw);
        }
        return balance;
    }
}

File: csharp/module-1/12_Polymorphism/exercise/Exercise/BankAccount.cs
namespace Exercise
{
    public class BankAccount
    {
        public string AccountHolderName { get; private set; }
        public string AccountNumber { get; }
        public decimal Balance { get; private set; }

        public BankAccount(string accountHolder, string accountNumber)
        {
            AccountHolderName = accountHolder;
            AccountNumber = accountNumber;
            Balance = 0;
        }

        public BankAccount(string accountHolder, string accountNumber, decimal balance)
        {
            AccountHolderName = accountHolder;
            AccountNumber = accountNumber;
            Balance = balance;
        }

        public decimal Deposit(decimal amountToDeposit)
        {
            if (amountToDeposit > 0)
            {
                Balance += amountToDeposit;
            }
            return Balance;
        }

        public virtual decimal Withdraw(decimal amountToWithdraw)
        {
            if (amountToWithdraw > 0)
            {
                Balance -= amountToWithdraw;
            }
            return Balance;
        }

    }
}
